meshtastic/node_runtime/seturl/rollback.py

```
@@ -0,0 +1,283 @@
+    """Owns rollback ordering/admin-index fallback and cache restore/invalidate strategy."""
+
+    def __init__(
+        self,
+        node: "Node",
+        *,
+        cache_manager: _SetUrlCacheManager,
+    ) -> None:
+        self._node = node
+        self._cache_manager = cache_manager
+
+    def rollback_add_only(
```

Ruff (0.15.7)

[warning] Too many branches (16 > 12)

(PLR0912)